.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,55 @@
+name: Run tests
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.6, 3.7]
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        python -m pip install --upgrade pip
+        hash -r
+        pip3 install -r pip-dep/requirements.txt
+        pip3 install -r pip-dep/requirements_dev.txt
+        pip3 install -r pip-dep/requirements_udacity.txt
+        pip3 install -r pip-dep/requirements_notebooks.txt
+        pip3 install scikit-learn
+        python setup.py install
+    - name: Lint, format, and type-check
```

Comment:
Not sure this will help to speed up things much, but what if we run linting tools deps installation?
Turnaround for formatting errors would be faster, in theory :)

Reply:
I agree!

Reply:
You might need to keep  `pip3 install -r pip-dep/requirements_dev.txt` however

Reply:
I was considering running linting as a completely separate action, since the first failure will cancel other actions in progress. Let's leave that for a future issue/PR though?